content/ADR-164-quests.poc.md

```diff
@@ -0,0 +1,211 @@
+
+For the sake of fast iteration and discovery, this document will explore the boundaries and scope of a Proof of Concept for the described Quests System.
+
+## Solution Space Exploration
+
+The following sections will describe what would be necessary and alternatives for this Proof of Concept, where designers can submit quests, users experience them and the system expose some metrics. 
+
+### Quest definition
+
+A Quest definition will include steps/tasks required for completion. It would be defined technically speaking as a graph, since it may have a non-linear complexity.
+
+Defining a valid graph, as it may require some conditions:
+- it should have only one start node.
+- it should have at least one end node.  
+- it may have cycles but not endless loops.
+- it may have conditional nodes.
+
+### Explorer
+
+Minimal UI for quest tracking and receive real-time notifications about the progress:
+
+- Users should have a UI where to see the current quests (in progress) and the status. 
+- Users should be able to cancel/drop a quest.
+- Users should be able to discover quests in World. (NPC Givers)
+
+This UI may be developed using SDK and defining reusable UI components so every Explorer implementation should not implement custom UIs.
+
+### Kernel
+
+Communication with the system, notifying events that happen in world, and receiving updates, so the Explorer is up to date.
+
+- Kernel should be responsible of receiving and notifiyng updates to Explorer.
+- Kernel should be able to receive requests from Explorer to start a quest and ask for it to the System.
+- Kernel should be able to notify in world events to the System, in order to make progress.
+
+### Actions
+
+A set of predefined actions may be provided to the designers and users, in order to define the Quests steps and make progress. 
+
+- Location
+- NPC Interaction
+- Play Emote
+- TBD
+
+### Quest Designer
+
+Build a tool/CLI to easily design Quests, without conflicting with existing definitions.
+
+Using the predefined actions, the designer should be able to create a Quest definition with steps and requirements. The output of a quest definition is a graph which will be persisted in a convenient form.
+
+### Backend
+
+The Quests System will expose an API with convenient endpoints for Designers and Users. 
+
+It also will process all events received from Explorer/Kernel sessions and make progress to corresponding Quest instances.
+
+Users may ask for quests state and metrics, but completed Quests may be stored for a certain time if the system needs space or it may grow undefinetly.
+
+- The system should be able to process all events in parallel, in order to make progress in corresponding Quests.
+- The system should be able to accept connections and subscriptions to Quest instances.
+- The system should be able to send updates to subscribed users.
+- The system should be able to respond to Quests state requests.
+- The system should expose metrics related to the Quests.
```

Comment:
Should we also add something about the system being able to call external (potentially allowlisted) services for rewards?